### PR TITLE
Document webpack config

### PR DESCRIPTION
**Pull request by MichaelGoberling**
This pull request proposes adding a guide for Webpack configurations. 

## Description

Add a guide explaining App Builder Webpack configuration options. 

## Related Issue

Closes https://github.com/AdobeDocs/project-firefly/issues/178

## Motivation and Context

To aid users in configuring Webpack with App Builder. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rperelma** [comment]
This is excellent, we receive so many questions about this topic!

File: src/pages/guides/index.md [hunk 1]
```diff
@@ -130,3 +130,6 @@ The [out-of-the-box CI/CD support](./deployment/ci_cd_for_firefly_apps.md) for A
 - [Github Workflows](https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow) to orchestrate the Github Actions upon specific events triggered against the application repository.
 - [Github Secrets](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets) to store application secrets required for the execution of the Github Workflows against specific environments.
 
+### Webpack 
+
+App Builder uses Webpack for bundling I/O Runtime action code. See [here](/guides/webpack-configuration) for an overview on configuring Webpack.
```
Comment (purplecabbage) [inline src/pages/guides/index.md:135]: I would only capitalize the first use of Webpack where you are speaking of the project.  The rest of the time I would just write webpack configuration, or webpack config ...

File: src/pages/guides/webpack-configuration.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
+# Webpack Configuration 
+
+## Overview
+
+App Builder bundles Runtime action code using Webpack. Users can specify a Webpack configuration for an individual action, a set of actions, 
+or an entire project. See [here](https://webpack.js.org/configuration/#options) for Webpack configuration options. 
+
+## Configuration File 
+
+When Runtime code is bundled, the CLI will search for a `*webpack-config.js` file starting at the path of the action code, working up to the root
```
Comment (purplecabbage) [inline src/pages/guides/webpack-configuration.md:10]: might want to make this more specific to the app-plugin the CLI doesn't really do much ...
When action code is bundled during `aio app build|run|deploy` the action directory is searched for a webpack config file named `*webpack-config.js`. The search is done first in the directory of the action being built, then the parent directory, up to the project root.

File: src/pages/guides/webpack-configuration.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+```
+
+### Base Options 
+
+App Builder starts off with the following Webpack configuration options, any additional values will be added on for these fields: 
+
+* `entry` - Will always contain the path to the action, additional paths identified in your configuration file will be added on.
+* `resolve.extensions` - Will always contain `.js` and `.json`, additional extensions identified in your configuration file will be added on.
+* `resolve.mainFields` - Will always contain `main`, additional main fields identified in your configuration file will be added on.
+* `plugins` - Will always contain a plugin that injects the following, additional plugins identified in your configuratoin file will be added on.
```
Comment (purplecabbage) [inline src/pages/guides/webpack-configuration.md:157]: This is really only used by our team, so we can deploy actions that are aware that they are running in a stage environment.  I think we can leave this detail out so people don't get confused by it.

File: src/pages/guides/webpack-configuration.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+* `output.path` - Will always be `./dist/${actionPath}/${actionName}-temp/`
+* `output.filename` - Will always be `${output.path}/index.js`
+  * Note: If supplying multiple configurations, this is true for the first configuration. For all subsequent configurations, it will be 
+    `${output.path}/index.[hashname].js`.  
+    
+## Environment Variables 
+
+If you have chosen to export a function, an async function, or an array of functions from your Webpack configuration file, App Builder will pass the environment to these functions. You can use this, for instance, to make decisions about bundling based on the environment. 
+
+Example: If you exported an envinronment variable called `MODE` before building your application, you can use this in your Webpack configuration file:
```
Comment (purplecabbage) [inline src/pages/guides/webpack-configuration.md:188]: This is good, might want to use something a little less generic like
process.env.FEATURE_FLAG_ENABLE_DONKEY

**Review by purplecabbage — APPROVED** [review]
This looks good to me, I added a few minor notes.

File: src/pages/guides/webpack-configuration.md (hunk 3 above)
Comment (shazron) [reply src/pages/guides/webpack-configuration.md:157]: spelling: `configuratoin`

File: src/pages/guides/webpack-configuration.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+   ```
+   {
+      WEBPACK_ACTION_BUILD: 'true', 
+      `process.env.AIO_CLI_ENV`: ${cliEnv}
+   }
+   ```
+   
+### Defaults 
+
+App Builder defaults the following Webpack configuration options if they are not present in your configuration file:
```
Comment (shazron) [inline src/pages/guides/webpack-configuration.md:168]: `App Builder defaults to the following Webpack`

File: src/pages/guides/webpack-configuration.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+App Builder defaults the following Webpack configuration options if they are not present in your configuration file: 
+
+* `output.libraryTarget` - Will default to `commonjs2`.
+* `mode` - Will default to `production`. 
+* `optimization.minimize` - Will default to `false`.
+
+### Immutable Options 
+
+App Builder does not allow changing the following Webpack configuration options. These fields will be ignored if present in your configuration file:
```
Comment (shazron) [inline src/pages/guides/webpack-configuration.md:176]: `App Builder does not allow changing of the following Webpack configuration options.`

**Review by shazron — CHANGES_REQUESTED** [review]

**Review by shazron — APPROVED** [review]

**Pull request merged**